exp/tests/test_contact_views.py

```diff
@@ -1,6 +1,285 @@
-# TODO: StudyParticipantContactView
-# - check can only GET if return user.is_researcher and user.has_study_perms(
-#             StudyPermission.CONTACT_STUDY_PARTICIPANTS, study
-#         )
-# - check context["participants"] has participants for this study but not another one
-# - check can post a single message (postpone checks of appropriate recipients & message type)
+import datetime
+
+from django.contrib.sites.models import Site
+from django.test import Client, TestCase, override_settings
+from django.urls import reverse
+from django_dynamic_fixture import G
+from guardian.shortcuts import assign_perm
+
+from accounts.backends import TWO_FACTOR_AUTH_SESSION_KEY
+from accounts.models import Child, Message, User
+from studies.models import Lab, Response, Study, StudyType
+from studies.permissions import StudyPermission
+
+
+class Force2FAClient(Client):
+    @property
+    def session(self):
+        _session = super().session
+        _session[TWO_FACTOR_AUTH_SESSION_KEY] = True
+        return _session
+
+
+# Run celery tasks right away, but don't catch errors from them.
+@override_settings(CELERY_TASK_ALWAYS_EAGER=True)
```

Comment:
Do we actually need these? The other thing we could do is use `mock.patch` decorator on the unit tests that call `some_task.delay` and check that the method was actually called (see [assert_called](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.assert_called)). In fact this is probably preferable; we have generally been bad about drawing distinctions between unit and integration tests but better start late than never.

Reply:
Done - very cool, I didn't realize we could patch `some_task.delay` so easily!